content/200-orm/200-prisma-client/150-using-raw-sql/100-typedsql.mdx

```diff
@@ -0,0 +1,204 @@
+---
+title: 'TypedSQL'
+metaTitle: 'Writing Type-safe SQL with TypedSQL and Prisma Client'
+metaDescription: 'Learn how to use TypedSQL to write fully type-safe SQL queries that are compatible with any SQL console and Prisma Client.'
+sidebar_class_name: preview-badge
+---
+
+## Getting started with TypedSQL
+
+To start using TypedSQL in your Prisma project, follow these steps:
+
+1. Ensure you have `@prisma/client` and `prisma` installed and updated to at least version `5.19.0`.
+
+1. Add the `typedSql` preview feature flag to your `schema.prisma` file:
+
+   ```prisma
+   generator client {
+     provider = "prisma-client-js"
+     previewFeatures = ["typedSql"]
+   }
+   ```
+
+1. Create a `sql` directory inside your `prisma` directory. This is where you'll write your SQL queries.
+
+   ```terminal
+   cd prisma
+   mkdir sql
+   ```
+
+1. Create a new `.sql` file in your `prisma/sql` directory. For example, `getUsersWithPosts.sql`. Note that the file name must be a valid JS identifier and cannot start with a `$`.
+
+1. Write your SQL queries in your new `.sql` file. For example:
+
+   ```sql title="prisma/sql/getUsersWithPosts.sql"
+   SELECT u.id, u.name, COUNT(p.id) as postCount
+   FROM users u
+   LEFT JOIN posts p ON u.id = p.authorId
+   GROUP BY u.id, u.name
+   ```
+
+1. Generate Prisma Client with the `sql` flag to ensure TypeScript functions and types for your SQL queries are created:
+
+   :::warning
+
+   Make sure that any pending migrations are applied before generating the client with the `sql` flag.
+
+   :::
+
+   ```terminal
+   prisma generate --sql
```

Comment:
Add a note to this step that `npx prisma generate --sql --watch` works as well please